src/main/scala/sigmastate/utils/SigmaByteWriter.scala

```diff
@@ -1,14 +1,63 @@
 package sigmastate.utils
 
-import scorex.util.ByteArrayBuilder
-import scorex.util.serialization.VLQByteBufferWriter
+import scorex.util.serialization.{ByteStringWriter, VLQByteBufferWriter, Writer}
+import scorex.util.serialization.Writer.Aux
 import sigmastate.SType
 import sigmastate.Values.Value
 import sigmastate.serialization.{ConstantStore, TypeSerializer, ValueSerializer}
 
-class SigmaByteWriter(b: ByteArrayBuilder,
-                      val constantExtractionStore: Option[ConstantStore]) extends VLQByteBufferWriter(b) {
+// todo rename to SigmaWriter?
+class SigmaByteWriter(val w: Writer,
```

Comment:
@ergomorphic Check out my first take on rewriting SigmaByteWriter to be a decorator for the underlying writer. All test are passing (except for new ergo serializer wrappers).